ruth/vehicle.py

```diff
@@ -22,6 +25,32 @@ def set_numpy_type(name, fld=None):
 IndexedNode = namedtuple("IndexedNode", ["node", "index"])
 
 
+class VehicleBehavior(IntFlag):
+    DEFAULT = 0
+    FASTEST_PATHS = auto()
+    SHORTEST_PATHS = auto()
+    DISTRIBUTED = auto()
+
+
+def set_vehicle_behavior(vehicles: List['Vehicle'], ratio: List[float], behavior: List[VehicleBehavior]):
+    """Set the behaviour of a given percentage of vehicles."""
+
+    assert isclose(sum(ratio), 1, abs_tol=1e-8)
+
+    n_vehicles = len(vehicles)
+    vehicles_shuffled = random.sample(vehicles, n_vehicles)
```

Comment:
I think that
```python
vehicles_shuffled = list(vehicles)
random.shuffle(vehicles_shuffled)
```
makes the intent more clear, I had to think about why is sampling used :) (as it emulates shuffling when `k=len(input)`).